application/liveness/liveness.go

```diff
@@ -0,0 +1,174 @@
+package liveness
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+	"encoding/csv"
+	"os"
+	"os/exec"
+)
+
+type LivenessTester interface {
+    PhantomIsLive(addr string, port uint16) (bool, error)
+}
+
+type CacheElement struct {
+    isLive 	bool
+	cachedTime	time.Time
+}
+
+type CachedLivenessTester struct{
+	ipCache 				map[string]CacheElement
+	signal 					chan bool
+	cacheExpirationTime 	float64

```

Comment:
It would be better if this were not float64 so we don't have to do time conversions when setting this. that seems relatively unnatural. I would prefer to make this a `time.Duration` object (here and in `Init()`). Then we can store the duration as a string and use [time.ParseDuration ](https://pkg.go.dev/time#ParseDuration). This would mean that the `CacheExpirationTime` would be a string and the base program can handle errors with parsing the time as a duration -- i.e. in `application/lib/register.go`

[short example](https://play.golang.org/p/nyuhXrxWhwF)